cupyx/scipy/special/convex_analysis.py

```diff
@@ -0,0 +1,129 @@
+from cupy import core
+
+
+_float_preamble = '''
+#ifndef NAN
+#define NAN __int_as_float(0x7fffffff)
+#endif
+
+#ifndef INF
+#define INF __int_as_float(0x7f800000)
+#endif
```

Comment:
How about using `CUDART_NAN` and `CUDART_INF`?
https://stackoverflow.com/a/15514595/402996

Reply:
Thanks, I was not aware of those. I copied the definition for NAN from the following location if you prefer to change it there as well:
https://github.com/cupy/cupy/blob/837d2a6108d9d58b4cf764cebac6f2036e35f3ad/cupy/math/misc.py#L96-L98